<launch>

  <!-- Bag -->
  <node name="rosbag" 
        pkg="rosbag" 
        type="play" 
        respawn="false"
        args="-s 0 -r 1.0 /home/ygling2008/testdata/icra2016rosbag/new/10.bag">
  </node>  

<!--args="-s 0 -r 1.0 /home/ygling2008/icra2016rosbag/new/8.bag"/-->
<!--args="-s 0 -r 1.0 /home/ygling2008/.ros/2015-11-24-12-07-59.bag"/-->

  <!-- Visual-inertial Estimator -->
  <node pkg="edge_imu_bias"
        name="edge_imu_bias"
        type="slam"       
        output="screen">
    <!--remap from="~image"    to="/camera/abc"/-->
    <param name="intrinsics_calibration_file"		value="$(find edge_imu_bias)/calib/combine2.yml"/>
    <param name="extrinsics_calibration_file"		value="$(find edge_imu_bias)/calib/visensor.yml"/>
    <param name="skipFrameNum"				value= "7"/>
    <param name="onUAV"					value= "false"/>
    <param name="enable_histogramEqualization"		value= "false"/>
    <param name="errorTranslationThreshold"		value= "0.3"/>
    <param name="errorAngleThreshold"			value= "0.1"/>
    <param name="adaptiveCannyThreshold"		value= "false"/>
    <param name="enable_LoopClosure"			value= "true"/>
    <param name="frontMarginalization"			value= "true"/>
    <param name="loopClosureInterval"			value= "0"/>
    <param name="visualWeight"				value= "100"/>
        <param name="max_feature_cnt" type="int" value="200" />
        <param name="num_of_iter" type="int" value="10" />
        <param name="calib_threshold_ric" type="double" value="0.0" />
        <param name="calib_threshold_tic" type="double" value="1.0" />
        <param name="init_depth" type="double" value="15.0" />
        <param name="gradient_threshold" type="double" value="1.0" />
        <param name="feature_threshold" type="double" value="5.0" />
        <param name="min_parallax" type="double" value="30.0" />
        <param name="min_parallax_point" type="double" value="5.0" />
        <param name="error_threshold" type="double" value="10000000000.0" />
        <param name="show_histogram" type="bool" value="false" />
        <param name="show_graph" type="bool" value="false" />
        <param name="show_html" type="bool" value="false" />
        <param name="multi_thread" type="bool" value="false" />
        <param name="imu_rate" type="double" value="200" />
        <param name="bias_acc_threshold" type="double" value="0.05" />
        <param name="bias_gyr_threshold" type="double" value="0.03" />
        <param name="solver_time" type="double" value="0.100" />
        <param name="compensate_rotation" type="bool" value="true" />
        <param name="acc_n" type="double" value="0.01" />
        <param name="acc_w" type="double" value="0.00002" />
        <param name="gyr_n" type="double" value="0.05" />
        <param name="gyr_w" type="double" value="4.0e-6" />
        <param name="ric_ok0" type="bool" value="true" />
        <param name="ric_y0" type="double" value="00.0" />
        <param name="ric_p0" type="double" value="0.0" />
        <param name="ric_r0" type="double" value="0.0" />
        <param name="tic_ok0" type="bool" value="true" />
        <param name="tic_x0" type="double" value="0.00" />
        <param name="tic_y0" type="double" value="0.06" />
        <param name="tic_z0" type="double" value="0.00" />
    <!--rosparam file="$(env PWD)/../bag/20150814-outdoor-stereo/calibration.yaml"/-->
  </node>  

</launch>
